x/blob/types/payforblob_test.go

```diff
@@ -174,29 +174,29 @@ func TestValidateBasic(t *testing.T) {
 			wantErr: nil,
 		},
 		{
-			name:    "parity shares namespace id",
+			name:    "parity shares namespace",
 			msg:     paritySharesMsg,
-			wantErr: ErrParitySharesNamespace,
+			wantErr: ErrInvalidBlobNamespace,
 		},
 		{
-			name:    "tail padding namespace id",
+			name:    "tail padding namespace",
 			msg:     tailPaddingMsg,
-			wantErr: ErrTailPaddingNamespace,
+			wantErr: ErrInvalidBlobNamespace,
 		},
 		{
-			name:    "transaction namspace namespace id",
+			name:    "tx namespace",
 			msg:     txNamespaceMsg,
-			wantErr: ErrReservedNamespace,
+			wantErr: ErrInvalidBlobNamespace,
```

Comment:
I could be wrong, but I think switching all of these to the same value result in users seeing a single error code for many different errors. That's ok, but what do we gain for wrapping each of these? Just that we see their path faster?

admittedly, I've just added errors to our giant list of official errors in the past, but perhaps we're at the point where documenting all of these is needed. For example, what is the difference between `ErrInvalidBlobNamespace`, `ErrInvalidNamespace`, `ErrTxNamespace`. I basically have to look at the code each time.

Reply:
You're right, after this PR user's will see a single error code: `11138` for every case with `ErrInvalidBlobNamespace`.

> what do we gain for wrapping each of these? Just that we see their path faster?

Hmm not sure what you mean by path. I could see an argument for preserving the old error codes and maintaining some level of duplication of `ValidateBlobNamespace()`.

I think the pain point here is that ideally we have functions exposed by the `namespace` package that return error strings and not Cosmos SDK errors (because the namespace package should remain sufficiently general). On the other hand, x/blob/types/payforblob.go should use custom defined Cosmos SDK errors so that the error codes and strings are unique and informative instead of the more general "invalid blob namespace". 

I'll take another stab at this.

Reply:
Took a different approach in https://github.com/celestiaorg/celestia-app/pull/2145/commits/14332d221fb0aed6f57ca08537d6395fc79cf521